### PR TITLE
Updated: OSS::Banner component with new size option + code cleanup

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?
Updated: OSS::Banner component with new size option + code cleanup

Screenshot 2023-11-28 at 11 34 52

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/banner.hbs [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
     <div>{{yield to="custom-icon"}}</div>
   {{else if @icon}}
     <OSS::Icon @style={{fa-icon-style @icon}} @icon={{fa-icon-value @icon}}
-               class="upf-badge upf-badge--shape-round upf-badge--size-md" />
+               class="upf-badge upf-badge--shape-round {{this.iconSizeClass}}" />
   {{else if @image}}
     <img class="upf-badge upf-badge--size-md upf-badge--shape-round" src={{@image}} alt="banner" />
   {{/if}}
-  <div class="fx-col fx-1 fx-gap-px-3">
+  <div class="fx-col fx-1 {{if (not-eq @size 'sm') 'fx-gap-px-3'}}">
```
Comment (JulienVannier66) [inline addon/components/o-s-s/banner.hbs:11]: Why remove the `gap-px-3` when the size is `sm`?

File: addon/components/o-s-s/banner.hbs [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
     {{#if @title}}
       <span class="font-weight-semibold text-size-5 font-color-gray-900">{{@title}}</span>
     {{/if}}
-    {{#if @subtitle}}
+    {{#if (has-block "secondary-actions")}}
+      <div>{{yield to="secondary-actions"}}</div>
+    {{else if @subtitle}}
```
Comment (JulienVannier66) [inline addon/components/o-s-s/banner.hbs:17]: You can't have `secondary-actions` block and subtitle together?

File: addon/components/o-s-s/banner.stories.js [hunk 3]
```diff
@@ -65,14 +66,25 @@ export default {
...
+    size: {
+      description:
+        'Allows to adjust the size of the component. Currently available options are `sm` and `md`. Defaults to `md`.',
+      table: {
+        type: { summary: 'string' },
+        defaultValue: { summary: 'false' }
+      },
+      control: {
+        type: 'string'
+      }
```
Comment (JulienVannier66) [inline addon/components/o-s-s/banner.stories.js:87]:
```suggestion
    size: {
      description:
        'Allows to adjust the size of the component. Currently available options are `sm` and `md`. Defaults to `md`.',
      table: {
        type: SizeTypes.join('|'),
        defaultValue: { summary: 'md' }
      },
      options: SizeTypes,
      control: {
        type: 'select'
      }
```
Nit: You can also define `const SizeTypes = ['sm', 'md'];` and set a selection control type for the doc. Wdyt?

File: addon/components/o-s-s/banner.hbs (hunk 1 above)
Comment (phndiaye) [inline addon/components/o-s-s/banner.hbs:11]: nit: `unless (eq @size "sm")`

File: addon/components/o-s-s/banner.hbs (hunk 2 above)
Comment (phndiaye) [inline addon/components/o-s-s/banner.hbs:17]: i wonder if we shouldn't have both subtitle AND `secondary-action` 🤔

File: addon/components/o-s-s/banner.stories.js (hunk 3 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/banner.stories.js:87]: for that little number of options you could also use a radio control type with`control: { type: 'radio' }`

File: addon/components/o-s-s/banner.hbs (hunk 1 above)
Comment (Miexil) [reply addon/components/o-s-s/banner.hbs:11]: because it's ugly :P

Comment (Miexil) [reply addon/components/o-s-s/banner.hbs:11]: without gap
Screenshot 2023-11-28 at 14 30 25

with gap
Screenshot 2023-11-28 at 14 30 40

File: addon/components/o-s-s/banner.hbs (hunk 2 above)
Comment (Miexil) [reply addon/components/o-s-s/banner.hbs:17]: No, you can't

Comment (Miexil) [reply addon/components/o-s-s/banner.hbs:17]: I guess you can, now ! :trollface:

File: addon/components/o-s-s/banner.stories.js (hunk 3 above)
Comment (Miexil) [reply addon/components/o-s-s/banner.stories.js:87]: type radio does not change anything from the type: select on my side 🤷

**Review by JulienVannier66 — APPROVED** [review]

**Review by phndiaye — APPROVED** [review]

**Pull request merged**